modules/apps/document-library/document-library-web-test/src/testIntegration/java/com/liferay/document/library/web/internal/product/navigation/control/menu/test/EditFileEntryHeaderProductNavigationControlMenuEntryTest.java

```diff
@@ -0,0 +1,212 @@
+	}
+
+	@Test
+	public void testIsShowFileEntryHeaderInAnotherView() throws Exception {
+		MockHttpServletRequest mockHttpServletRequest =
+			_getMockHttpServletRequest(_createFileEntry());
+
+		mockHttpServletRequest.setParameter("mvcRenderCommandName", "/");
+
+		Assert.assertFalse(
+			_productNavigationControlMenuEntry.isShow(mockHttpServletRequest));
+	}
+
+	@Test
+	public void testIsShowFileEntryHeaderInViewerView() throws Exception {
+		FileEntry fileEntry = _createFileEntry();
+
+		MockHttpServletRequest mockHttpServletRequest =
+			_getMockHttpServletRequest(fileEntry);
+
+		mockHttpServletRequest.setParameter(
+			"mvcRenderCommandName", "/document_library/view_file_entry");
+
+		_deleteResourcePermissions(fileEntry);
+
+		Assert.assertTrue(
+			_productNavigationControlMenuEntry.isShow(mockHttpServletRequest));
+	}
+
+	@Test
+	public void testIsShowFileEntryHeaderWithGuestPermissions()
+		throws Exception {
+
+		Assert.assertFalse(
+			_productNavigationControlMenuEntry.isShow(
+				_getMockHttpServletRequest(_createFileEntry())));
+	}
+
+	@Test
+	public void testIsShowFileEntryHeaderWithoutFileEntry() throws Exception {
+		Assert.assertFalse(
+			_productNavigationControlMenuEntry.isShow(
+				_getMockHttpServletRequest(null)));
+	}
+
+	@Test
+	public void testIsShowFileEntryHeaderWithoutGuestPermissions()
+		throws Exception {
+
+		FileEntry fileEntry = _createFileEntry();
+
+		_deleteResourcePermissions(fileEntry);
+
+		Assert.assertTrue(
+			_productNavigationControlMenuEntry.isShow(
+				_getMockHttpServletRequest(fileEntry)));
+	}
+
+	private FileEntry _createFileEntry() throws Exception {
+		return _dlAppService.addFileEntry(
+			null, _group.getGroupId(),
+			DLFolderConstants.DEFAULT_PARENT_FOLDER_ID,
+			RandomTestUtil.randomString(),
+			MimeTypes.MIME_APPLICATION_OCTET_STREAM,
+			RandomTestUtil.randomString(), null, RandomTestUtil.randomString(),
+			StringPool.BLANK, (byte[])null, null, null, null,
+			ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+	}
+
+	private void _deleteResourcePermissions(FileEntry fileEntry)
+		throws Exception {
+
+		_resourcePermissionLocalService.deleteResourcePermissions(
+			_group.getCompanyId(), DLFileEntry.class.getName(),
+			ResourceConstants.SCOPE_INDIVIDUAL, fileEntry.getFileEntryId());
+	}
+
+	private MockHttpServletRequest _getMockHttpServletRequest(
+			FileEntry fileEntry)
+		throws Exception {
+
+		MockHttpServletRequest mockHttpServletRequest =
+			new MockHttpServletRequest();
+
+		mockHttpServletRequest.addParameter(
+			"mvcRenderCommandName", "/document_library/edit_file_entry");
+		mockHttpServletRequest.setAttribute(
+			JavaConstants.JAVAX_PORTLET_RESPONSE,
+			new MockLiferayResourceResponse());
+
+		if (fileEntry != null) {
+			mockHttpServletRequest.setAttribute(
+				WebKeys.DOCUMENT_LIBRARY_FILE_ENTRY, fileEntry);
+		}
+
+		mockHttpServletRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, _getThemeDisplay());
+
+		return mockHttpServletRequest;
```

Comment:
You need to set the attributes in alphabetical order: DOCUMENT_LIBRARY_FILE_ENTRY < JAVAX_PORTLET_RESPONSE < THEME_DISPLAY

Reply:
Hi @adolfopa ,
I sent the changes. Thank you!